core/src/com/unciv/logic/civilization/diplomacy/DiplomacyManager.kt

```diff
@@ -714,12 +729,34 @@ class DiplomacyManager() : IsPartOfGameInfoSerialization {
     }
 
     /** Everything that happens to both sides equally when war is declared by one side on the other */
-    private fun onWarDeclared() {
+    private fun onWarDeclared(isOffensiveWar: Boolean) {
         // Cancel all trades.
         for (trade in trades)
             for (offer in trade.theirOffers.filter { it.duration > 0 })
                 civInfo.addNotification("[${offer.name}] from [$otherCivName] has ended",
                     NotificationCategory.Trade, otherCivName, NotificationIcon.Trade)
+
+        // If we attacked, then we need to end all of our defensive pacts acording to Civ 5
+        if (isOffensiveWar) {
+            civInfo.diplomacy.values.filter { it.otherCiv() != otherCiv() // So that we don't cancel the defensive pact before negative modifiers are added
+                && it.diplomaticStatus == DiplomaticStatus.DefensivePact }.forEach {
+                // Trades with defensive pact are now invalid
+                val defensivePactOffer = it.trades.firstOrNull { trade -> trade.ourOffers.any { offer -> offer.name == Constants.defensivePact} }
+                it.trades.remove(defensivePactOffer)
+                val theirDefensivePactOffer = it.otherCivDiplomacy().trades.firstOrNull { trade -> trade.ourOffers.any { offer -> offer.name == Constants.defensivePact} }
+                it.otherCivDiplomacy().trades.remove(theirDefensivePactOffer)
+                it.removeFlag(DiplomacyFlags.DefensivePact)
+                it.otherCivDiplomacy().removeFlag(DiplomacyFlags.DefensivePact)
+                it.diplomaticStatus = DiplomaticStatus.Peace
+                it.otherCivDiplomacy().diplomaticStatus = DiplomaticStatus.Peace
+                for (civ in getCommonKnownCivs().filter { civ -> civ.isMajorCiv() }) {
+                    civ.addNotification("[${civInfo.civName}] canceled their Defensive Pact with [${it.otherCivName}]!",
+                        NotificationCategory.Diplomacy, civInfo.civName, NotificationIcon.Diplomacy, it.otherCivName)
+                }
+                civInfo.addNotification("We have canceled our Defensive Pact with [${it.otherCivName}]!",
```

Comment:
What about the other side? Looks like they don't get a notification, only for us and 3rd parties

Reply:
I experimented with it, but getCommonKnownCivs() actually returns the other civ as well. It doesn't, however return the Civ that this DiplomacyManager belongs to. Logically it doesn't make sense for GetCommonKnownCivs() to return the other Civ as well and not the current civ but that is how it functions aparently.

Reply:
That's very strange, I'm pretty sure there's a test for that
In any case that's something we need to fix

Reply:
No, I'm certain there are tests for that
I'll recheck this to make sure

Reply:
I figured it out. I saw the notification from the trade. There was no notification from it. I'll change it to not send a trade-end notification.